### ISSUE TITLE
how should I deal with Depwise batchnormalization

### ISSUE THREAD

**Issue by kanbaiou** [google-research/morph-net#32]
thank you  for your excellent work. 
Here I decide to realize this algorithm by myself according to this paper in MobileNetV2 ,and met one problems.
In the case of using model_size regularizer ,  C_cost of every layer should be computed to define the cost function.And we train the model to decide how to shrink the Network.

But I notice that in MobileNetV2 , the input and output channels should be kept  same in depthwise conv-layer. Is this means   I should ignore this layer when shrink the model? 
and in this case when computing the lambda  in cost function, I should also ignore this part's model size. 

thank you very much.

**Comment by eladeban** [comment]
As far as I know the ModelSize regularizer works properly on MobileNetV2. 

As you point out, the correct behavior is to **not** regularize the batch norm on the depth-wise convolutions, as these operations cannot be controlled in terms of changing their number of outputs. Essentially, a depth-wise convolution is more like a polling layer than a convolution for our algorithm.

With that said, to have an accurate regularizer, the cost (model size or flops) of the depth-wise conv should be pushed into the cost of the preceding op (which is also handled by our framework).

**Comment by kanbaiou** [comment]
Got this, thank you very much !
